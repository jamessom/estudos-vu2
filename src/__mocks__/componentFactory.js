import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueRouter from '@/router/index';

export default function componentFactory(options = {}) {
  return {
    wrapper: () => {
      const { component, componentOptions, withRouter } = options;
      const localvue = createLocalVue();

      if (withRouter) {
        localvue.use(VueRouter);

        return shallowMount(component, {
          localvue,
          ...componentOptions,
        });
      }

      return shallowMount(component, {
        localvue,
        ...componentOptions,
      });
    },
  };
}
