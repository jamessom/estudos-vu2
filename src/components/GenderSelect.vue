<template>
  <div class="field">
    <label class="label">{{ this.label }}</label>

    <div class="control is-expanded">
      <div class="select is-fullwidth">
        <select v-model="selected_option">
          <option
            v-for="option in this.options"
            :value="option.value"
            :key="option.index">
              {{option.text}}
          </option>
        </select>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'GenderSelect',
  props: {
    label: String,
    options: Array,
  },
  data() {
    return {
      selected_option: 'selected',
    };
  },
};
</script>
