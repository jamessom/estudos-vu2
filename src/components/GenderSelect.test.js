import componentFactory from '@/__mocks__/componentFactory';
import faker from 'faker';
import GenderSelect from './GenderSelect.vue';

describe('src/components/GenderSelect.vue', () => {
  describe('when component is mounted', () => {
    const snapshotTest = componentFactory({ component: GenderSelect });

    it('should renders correctly', () => {
      expect(snapshotTest.wrapper().element).toMatchSnapshot();
    });
  });

  describe('create components and change default props', () => {
    const propsData = {
      label: faker.random.word(),
      options: [
        { value: faker.random.word(), text: faker.random.word() },
        { value: faker.random.word(), text: faker.random.word() },
        { value: faker.random.word(), text: faker.random.word() },
      ],
    };

    const factory = componentFactory({
      component: GenderSelect,
      componentOptions: {
        propsData,
      },
    });

    it('should create a select component with label text', () => {
      expect(factory.wrapper().props().label).toBe(propsData.label);
    });

    it('should create a select component with options', () => {
      expect(factory.wrapper().props().options).toBe(propsData.options);
    });


    describe('options must be an array', () => {
      it('should have a title', () => {
        expect(factory.wrapper().props().options[0]).toHaveProperty(
          'title',
          propsData.options[0].title,
        );
      });

      it('should have a value', () => {
        expect(factory.wrapper().props().options[0]).toHaveProperty(
          'value',
          propsData.options[0].value,
        );
      });
    });
  });
});
