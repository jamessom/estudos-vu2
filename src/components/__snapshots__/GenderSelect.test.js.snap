// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/GenderSelect.vue when component is mounted should renders correctly 1`] = `
<div
  class="field"
>
  <label
    class="label"
  >
    
  </label>
   
  <div
    class="control is-expanded"
  >
    <div
      class="select is-fullwidth"
    >
      <select />
    </div>
  </div>
</div>
`;
